ui/packages/app/src/project_setting/project_info/ProjectInfoForm.js

```diff
@@ -0,0 +1,121 @@
+  const { project, setStream, setTeam, setLabels } = useContext(
+    ProjectFormContext
+  );
+
+  const [isValidStream, setIsValidStream] = useState(
+    isDNS1123Label(project.stream)
+  );
+  const [isValidTeam, setIsValidTeam] = useState(isDNS1123Label(project.team));
+
+  useEffect(() => {
+    if (!project.team) {
+      setIsValidTeam(false);
+    }
+  }, [project.team]);
+
+  const [isValidLabels, setIsValidLabels] = useState(
+    project.labels.length === 0
+      ? true
+      : project.labels.reduce((labelsValid, label) => {
+          return (
+            labelsValid &&
+            isDNS1123Label(label.key) &&
+            isDNS1123Label(label.value)
+          );
+        }, true)
+  );
+
+  const isDisabled = !config.METADATA_UPDATE_ENABLED;
+
+  return (
+    <>
+      <EuiFlexGroup direction="column">
+        <EuiForm>
+          <EuiFlexItem>
+            <EuiTitle size="s">
+              <h3>Stream</h3>
+            </EuiTitle>
+            <EuiSpacer size="m" />
+            <EuiText size="s" color="subdued">
+              <p>Product stream the project belongs to</p>
+            </EuiText>
+            <EuiSpacer size="m" />
+            <Stream
+              stream={project.stream}
+              setStream={setStream}
+              setIsValidStream={setIsValidStream}
+              isDisabled={isDisabled}
+            />
+            <EuiSpacer size="xl" />
+            <EuiTitle size="s">
+              <h3>Team</h3>
+            </EuiTitle>
+            <EuiSpacer size="m" />
+            <EuiText size="s" color="subdued">
+              <p>The owner of the project</p>
+            </EuiText>
+            <EuiSpacer size="m" />
+            <Team
+              team={project.team}
+              setTeam={setTeam}
+              stream={project.stream}
+              setIsValidTeam={setIsValidTeam}
+              isDisabled={isDisabled}
+            />
+            <EuiSpacer size="xl" />
+            <EuiTitle size="s">
+              <h3>Labels</h3>
+            </EuiTitle>
+            <EuiSpacer size="m" />
+            <EuiText size="s" color="subdued">
+              <p>Additional Labels</p>
+            </EuiText>
+            <EuiSpacer size="m" />
+            <Labels
+              labels={project.labels}
+              setLabels={setLabels}
+              setIsValidLabels={setIsValidLabels}
+              isValidLabels={isValidLabels}
+              isDisabled={isDisabled}
+            />
+          </EuiFlexItem>
+          <EuiSpacer size="m" />
+          <EuiFlexItem>
+            <SubmitProjectInfoForm
+              project={project}
+              fetchUpdates={fetchUpdates}
+              isValidTeam={isValidTeam}
+              isValidStream={isValidStream}
+              isValidLabels={isValidLabels}
+              selectedStream={project.stream}
+              selectedTeam={project.team}
+              selectedLabels={project.labels}
```

Comment:
Since we are already passing `project`, which contains the stream, team and labels, as a prop to `SubmitProjectInfoForm`, can we reduce the number of additional props in `SubmitProjectInfoForm` and `UpdateProjectInfoModal`?